OpenSim/Actuators/ActiveForceLengthCurve.cpp

```diff
@@ -153,15 +153,26 @@ void ActiveForceLengthCurve::setMinValue(double minimumValue)
 //==============================================================================
 double ActiveForceLengthCurve::calcValue(double normFiberLength) const
 {
-    SimTK_ASSERT(isObjectUpToDateWithProperties(),
+    OPENSIM_ASSERT(
+        isObjectUpToDateWithProperties() &&
```

Comment:
Here and elsewhere: I don't understand why this assertion is written as a logical 'and' with `isObjectUpToDateWithProperties()` and a string. Simpler would be easier to read. Just delete the string since it always evaluates to false?

Reply:
This was new to me, but @adamkewley pointed out to me that a common paradigm in c++ is to have `assert(condition && "this condition failed")`, which indeed evaluates the string as false only if the condition is false.

It is also listed as a common way to display error messages: https://en.cppreference.com/w/cpp/error/assert

Reply:
Ah, ok thanks. Is this now the preferred style (supplanting `SimTK_ASSERT` and `OPENSIM_ASSERT_FRMOBJ`)?

Reply:
Yeah, afaik: https://github.com/opensim-org/opensim-core/pull/3653#discussion_r1426588089